packages/compass-e2e-tests/tests/oidc.test.ts

```diff
@@ -0,0 +1,168 @@
+import type { Compass } from '../helpers/compass';
+import type { OIDCMockProviderConfig } from '@mongodb-js/oidc-mock-provider';
+import { OIDCMockProvider } from '@mongodb-js/oidc-mock-provider';
+import type { ChildProcess } from 'child_process';
+import { spawn } from 'child_process';
+import path from 'path';
+import os from 'os';
+import { promises as fs } from 'fs';
+import { getDownloadURL } from 'mongodb-download-url';
+import tar from 'tar';
+import { promisify } from 'util';
+import type { Readable } from 'stream';
+import { pipeline, PassThrough } from 'stream';
+import { createInterface as readline } from 'readline';
+import https from 'https';
+import { once } from 'events';
+import { expect } from 'chai';
+
+describe('OIDC integration', function () {
+  let compass: Compass;
+  let browser: CompassBrowser;
+
+  let getTokenPayload: typeof oidcMockProviderConfig.getTokenPayload;
+  let oidcMockProviderConfig: OIDCMockProviderConfig;
+  let oidcMockProvider: OIDCMockProvider;
+
+  let i = 0;
+  let tmpdir: string;
+  let server: ChildProcess;
+  let serverExit: Promise<unknown>;
+  let connectionString: string;
+
+  before(async function () {
+    // TODO(MONGOSH-1306): Get rid of all the setup code to download mongod here... :(
+    if (process.platform !== 'linux') {
+      return this.skip();
+    }
+
+    {
+      oidcMockProviderConfig = {
+        getTokenPayload(metadata: Parameters<typeof getTokenPayload>[0]) {
+          return getTokenPayload(metadata);
+        },
+      };
+      oidcMockProvider = await OIDCMockProvider.create(oidcMockProviderConfig);
+    }
+
+    {
+      tmpdir = path.join(
+        os.tmpdir(),
+        `compass-oidc-${Date.now().toString(32)}-${++i}`
+      );
+      await fs.mkdir(path.join(tmpdir, 'db'), { recursive: true });
+    }
+
+    {
+      const { url } = await getDownloadURL({
+        version: '>= 7.0.0-rc0',
+        enterprise: true,
+      });
+
+      await promisify(pipeline)(
+        await new Promise<Readable>((resolve) => https.get(url, resolve).end()),
+        tar.x({ cwd: tmpdir, strip: 1 })
+      );
+    }
+
+    {
+      const serverOidcConfig = {
+        issuer: oidcMockProvider.issuer,
+        clientId: 'testServer',
+        requestScopes: ['mongodbGroups'],
+        authorizationClaim: 'groups',
+        audience: 'resource-server-audience-value',
+        authNamePrefix: 'dev',
+      };
+
+      server = spawn(
+        path.join(tmpdir, 'bin', 'mongod'),
+        [
+          '--setParameter',
+          'authenticationMechanisms=SCRAM-SHA-256,MONGODB-OIDC',
+          // enableTestCommands allows using http:// issuers such as http://localhost
+          '--setParameter',
+          'enableTestCommands=true',
+          '--setParameter',
+          `oidcIdentityProviders=${JSON.stringify([serverOidcConfig])}`,
+          '--dbpath',
+          path.join(tmpdir, 'db'),
+          '--port',
+          '0',
```

Comment:
TIL port=0 makes it find a random (unused?) port. nice

Reply:
Yeah, it’s a nice feature of the OS and it’s good that the server also allows specifying this. Ideally we’ll want to use this for our mongodb-runner/mlaunch replacement so that we’ll just never have to worry about port conflicts.